File: test/fixtures/sample.py
#!/usr/bin/env python3
"""
Sample Python file for LSP testing.
Tests diagnostics, completion, hover, and formatting.
"""

import os
import sys
from typing import List, Dict, Optional


class Calculator:
    """A simple calculator class for testing."""

    def __init__(self, initial_value: float = 0):
        self.value = initial_value
        self._history: List[float] = [initial_value]

    def add(self, amount: float) -> 'Calculator':
        """Add amount to current value."""
        self.value += amount
        self._history.append(self.value)
        return self

    def multiply(self, factor: float) -> 'Calculator':
        """Multiply current value by factor."""
        self.value *= factor
        self._history.append(self.value)
        return self

    def get_history(self) -> List[float]:
        """Return calculation history."""
        return self._history.copy()

    # Intentional error for diagnostic testing
    def bad_method(self):
        undefined_variable  # This should trigger a diagnostic
        return self.value / 0  # This too


def process_data(data: List[int],
                 processor: Optional[callable] = None) -> Dict[str, any]:
    """Process a list of integers with optional processor function."""
    if processor is None:
        processor = lambda x: x * 2

    results = {
        'original': data,
        'processed': [processor(item) for item in data],
        'sum': sum(data),
        'avg': sum(data) / len(data) if data else 0
    }

    # Test completion here: os.path.
    # Test hover on any parameter

    return results


# Formatting test - this should be reformatted
def poorly_formatted(x, y, z):
    return x + y + z


# Code action test - missing import
def use_pandas():
    df = pd.DataFrame()  # pandas not imported
    return df


if __name__ == "__main__":
    calc = Calculator(10)
    result = calc.add(5).multiply(2).get_history()
    print(f"Calculator history: {result}")

    data = [1, 2, 3, 4, 5]
    processed = process_data(data)
    print(f"Processed data: {processed}")

    # Another intentional error
    print(undefined_function())

File: test/fixtures/sample.lua
-- Sample Lua file for LSP testing

local M = {}

-- Simple function with documentation
---@param x number First number
---@param y number Second number
---@return number result The sum of x and y
function M.add(x, y)
    return x + y
end

-- Function with table parameter
---@param config table Configuration options
---@field name string The name
---@field enabled boolean Whether enabled
---@return boolean success
function M.configure(config)
    if not config.name then
        return false
    end

    -- Incomplete line for completion testing
    local result = config.

    return config.enabled or false
end

-- Class-like structure
---@class Calculator
---@field value number Current value
local Calculator = {}
Calculator.__index = Calculator

---@return Calculator
function Calculator.new()
    local self = setmetatable({}, Calculator)
    self.value = 0
    return self
end

---@param amount number
function Calculator:add(amount)
    self.value = self.value + amount
end

-- Intentional errors for diagnostic testing
function M.broken_function()
    local undefined_var = unknown_global

    -- Type mismatch
    local str = "hello"
    local num = str + 5  -- Error: attempt to perform arithmetic on string

    -- Missing 'then'
    if true
        print("missing then")
    end
end

-- Table with methods for completion testing
M.utils = {
    format = function(str) return string.format("%s", str) end,
    split = function(str, sep)
        -- Incomplete for testing
        return vim.
    end,
}

-- Test requiring another module
local status_ok, telescope = pcall(require, "telescope")
if status_ok then
    -- Test telescope completion
    telescope.
end

return M
